### PR TITLE
fixes #8

### LINKED ISSUE #8: Unable to load ethtool plugin
@lmroz asked for stderr and ethtool results:

[snap-plugin-collector-ethtool.stderr.txt](https://github.com/intelsdi-x/snap-plugin-collector-ethtool/files/186040/snap-plugin-collector-ethtool.stderr.txt)

[ethtool.txt](https://github.com/intelsdi-x/snap-plugin-collector-ethtool/files/186055/ethtool.txt)

root@node-1:~# ethtool -i eth0
driver: ixgbe
version: 3.15.1-k
firmware-version: 0x80000647
bus-info: 0000:01:00.0
supports-statistics: yes
supports-test: yes
supports-eeprom-access: yes
supports-register-dump: yes
supports-priv-flags: no

root@node-1:~# ethtool -i eth1
driver: ixgbe
version: 3.15.1-k
firmware-version: 0x80000647
bus-info: 0000:01:00.1
supports-statistics: yes
supports-test: yes
supports-eeprom-access: yes
supports-register-dump: yes
supports-priv-flags: no

### PR DESCRIPTION
**Pull request by lmroz**
Summary of changes:

- Patched GetDriverInfo
- Covered it by tests

Testing done:
- Unit tests

### CODE REVIEW

File: etplugin/metric_collector.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func (mc *metricCollectorImpl) ValidMetrics() (map[metricSource][]string, error)
 
 		driverName, err := mc.Ethtool.GetDriverInfo(iface.Name)
 		if err != nil {
-			return nil, err
+			continue
 		}
```
Comment (mkuculyma) [inline etplugin/metric_collector.go]: Small detail:
This change causes error message returned by GetDriverInfo() (https://github.com/lmroz/snap-plugin-collector-ethtool/blob/master/ethtool/collector.go#L58) to be never used. Consider logging the message in GetDriverInfo() or maybe remove it.

File: etplugin/plugin.go [hunk 2]
```diff
@@ -29,7 +29,7 @@ const (
 	// Name of plugin
 	Name = "ethtool"
 	// Version of plugin
-	Version = 4
+	Version = 5
 	// Type of plugin
```
Comment (mkuculyma) [inline etplugin/plugin.go]: You could also update godeps.

File: ethtool/collector.go [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /*
 http://www.apache.org/licenses/LICENSE-2.0.txt
-Copyright 2015 Intel Corporation
+Copyright 2016 Intel Corporation
```
Comment (IzabellaRaulin) [inline ethtool/collector.go]: Please, do not change the date in that way, leave `2015` as the date of creation the file was created.

File: etplugin/plugin.go (hunk 2 above)
Comment (lmroz) [inline etplugin/plugin.go]: fixed

**Comment by katarzyna-z** [comment]
LGTM

**Pull request merged**